### ISSUE TITLE
Minimap2 problem with novel transcriptome

### ISSUE THREAD

**Issue by matthew-valentine** [lh3/minimap2#957]
I am running into an issue when trying to map ONT long-read data to a custom transcriptome. The transcriptome in question is a mix of GENCODE transcripts, plus some novel transcripts from using TALON for isoform discovery. As part of the TALON pipeline I mapped the samples to the GENCODE transcripts alone, and found around 2% of the reads couldn't map. When I take these same GENCODE transcripts, but add in the novel transcripts found from TALON, the unmapped reads jump up to ~20%.

In testing, I separated out the unmapped reads, and then tried remapping to GENCODE and TALON transcripts separately. In both cases I am able to rescue a lot of the unmapped reads. I can include an example read below:

GENCODE MAPPING:

`91:312|c358c398-901c-4661-bafe-b917a388c0ee     0       ENST00000616345.1|ENSG00000275616.1|-|-|U2.11-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       ATCGCTTCTCGGCCTTTTGGCTAAGATCAAGTGTAGTATCTGTTTCTTATCAGTTTAATATCTGATACGTCCTCTATCCGAGGACAATATATTAAATGGATTTTTGGAGCAGGGAGATGGAATAGGAGCTTGCTCCGTCCACTCCACGCATCGACCTGGTATTGCAGTACCTCCAGGAACGGTGCACC
    0166643677;;;<;>CA?>;;<;8<=<?736:;:99;=@>==>*,ABB@@4454.++,:>A7667=76669;;=>AA75448>>>>=22{0001147()677=87:877,++/.==>??=;60..,++*-*)*+4455686876552223:4479::;;::;::8669;=<=<7778?9///0>987    NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:P  cm:i:33 s1:i:183        s2:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000615427.1|ENSG00000274862.1|-|-|U2.9-201|U2|191|snRNA|  1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000620268.1|ENSG00000275219.1|-|-|U2.10-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000613778.1|ENSG00000276596.1|-|-|U2.12-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000611544.1|ENSG00000277903.1|-|-|U2.15-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000619465.1|ENSG00000274432.1|-|-|U2.5-201|U2|191|snRNA|  1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000613119.1|ENSG00000278048.1|-|-|U2.16-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000617785.1|ENSG00000274062.1|-|-|U2.3-201|U2|191|snRNA|  1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000618664.1|ENSG00000274585.1|-|-|RNU2-1-201|RNU2-1|191|snRNA|    1       0       42M1I145M       *       0       0
       *       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000618602.1|ENSG00000278591.1|-|-|U2.19-201|U2|191|snRNA| 1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000616535.1|ENSG00000273709.1|-|-|U2.2-201|U2|191|snRNA|  1       0       42M1I145M       *       0       0       *
       *       NM:i:1  ms:i:368        AS:i:368        nn:i:0  ts:A:+  tp:A:S  cm:i:33 s1:i:183        de:f:0.0053     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000613956.1|ENSG00000278774.1|-|-|U2.20-201|U2|192|snRNA| 1       0       5M1D37M1I145M   *       0       0       *
       *       NM:i:2  ms:i:362        AS:i:362        nn:i:0  ts:A:+  tp:A:S  cm:i:32 s1:i:178        de:f:0.0106     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000619225.1|ENSG00000274452.1|-|-|U2.6-201|U2|191|snRNA|  1       0       42M1I145M       *       0       0       *
       *       NM:i:2  ms:i:362        AS:i:362        nn:i:0  ts:A:+  tp:A:S  cm:i:29 s1:i:167        de:f:0.0106     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000538098.2|ENSG00000133316.16|OTTHUMG00000167695.4|OTTHUMT00000396014.2|WDR74-208|WDR74|548|protein_coding|      1
       0       42M1I144M1S     *       0       0       *       *       NM:i:6  ms:i:336        AS:i:336        nn:i:0  ts:A:+  tp:A:S  cm:i:27 s1:i:158        de:f:0.0321     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000525239.5|ENSG00000133316.16|OTTHUMG00000167695.4|OTTHUMT00000395678.1|WDR74-204|WDR74|1736|protein_coding|     1
       0       42M1I144M1S     *       0       0       *       *       NM:i:6  ms:i:336        AS:i:336        nn:i:0  ts:A:+  tp:A:S  cm:i:27 s1:i:158        de:f:0.0321     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     ENST00000410396.1|ENSG00000222328.1|-|-|RNU2-2P-201|RNU2-2P|191|snRNA|  1       0       42M1I144M1S     *       0       0
       *       *       NM:i:6  ms:i:336        AS:i:336        nn:i:0  ts:A:+  tp:A:S  cm:i:27 s1:i:158        de:f:0.0321     rl:i:0
       `

TALON MAPPING:

`91:312|c358c398-901c-4661-bafe-b917a388c0ee     0       TALONT000270875 1       60      42M1I144M1S     *       0       0       ATCGCTTCTCGGCCTTTTGGCTAAGATCAAGTGTAGTATCTGTTTCTTATCAGTTTAATATCTGATACGTCCTCTATCCGAGGACAATATATTAAATGGATTTTTGGAGCAGGGAGATGGAATAGGAGCTTGCTCCGTCCACTCCACGCATCGACCTGGTATTGCAGTACCTCCAGGAACGGTGCACC    0166643677;;;<;>CA?>;;<;8<=<?736:;:99;=@>==>*,ABB@@4454.++,:>A7667=76669;;=>AA75448>>>>=22{0001147()677=87:877,++/.==>??=;60..,++*-*)*+4455686876552223:4479::;;::;::8669;=<=<7778?9///0>987    NM:i:6  ms:i:336        AS:i:336
        nn:i:0  ts:A:+  tp:A:P  cm:i:27 s1:i:158        s2:i:147        de:f:0.0321     rl:i:0
91:312|c358c398-901c-4661-bafe-b917a388c0ee     256     TALONT000587995 1       0       42M1I115M30S    *       0       0       *       *       NM:i:5  ms:i:284        AS:i:284
        nn:i:0  ts:A:+  tp:A:S  cm:i:24 s1:i:147        de:f:0.0316     rl:i:0
`

It seems the mapping for the TALON transcripts is similar to the mapping for some of the secondary alignments in GENCODE (same CIGAR and alignment score). I would have expected when having both the GENCODE and TALON transcripts available that this read would have 18 alignments (all of the GENCODE only ones plus the two TALON only ones) but in fact it is marked as unmapped. Any idea why this would be the case? I originally thought maybe there was some threshold for number of different transcripts a read can align to before it is discarded (as in STAR), but looking at examples like this, where adding the novel transcripts would only lead to two additional mapping events, I am assuming this isn't what is happening.

For reference, this is the code I used for mapping, and I am using v2.17

`minimap2 -t 1 -2 -ax map-ont -p 1 -N 100 -k15 -uf [transcripts fasta.mmi] [input file] > out.sam`

**Comment by lh3** [comment]
Try the latest version and increase the k-mer frequency threshold to something like `-f 1000` or above.

**Comment by matthew-valentine** [comment]
Thanks for the advice. I gave this a go with v2.24 and I was able to rescue these unmapped reads using the combined transcriptome for mapping. One thing that I don't quite understand though: when mapping to the combined transcriptome, for the example read I showed above it now maps to 101 loci, but it doesn't map to either of the TALON transcripts it mapped to before. I thought maybe it was related to increasing this -f parameter, but when I mapped to GENCODE and TALON separately with -f 1000, I got the same mapping results as before, for a grand total of 18 mapped loci. Is there a reason that I am getting so much more mapping when both sets of transcripts are present than when running them separately?
